### ISSUE TITLE
shipyard/outfitter info is only available for visited planets, not revealed systems

### ISSUE THREAD

**Issue by flowers9** [endless-sky/endless-sky#9050]
### Problem Description

Currently, on the MapShipyardPanel and MapOutfitterPanel, stocking information for shipyards and outfitters is displayed for any system that has been revealed on the map by either visiting it or by buying a map from an outfitter.  However, this can include information for hostile or secretive aliens that you can't even communicate with, which makes little sense and clutters the display with ships or outfits that can't be purchased in the forseeable future.

### Related Issue Links

I haven't run across any other discussion of this.

### Desired Solution

I'd suggest changing it to display information for visited planets instead.  In fact, I'd always assumed that was how the game worked and made a point of visiting planets with shipyards/outfitters just to collect the information.  It wasn't until I realized that outfitter maps gave the information for remote system that I realized that wasn't how it worked.

While this is mostly a trivial code change, it creates an issue with the MapSalesPanel, which displays the number of ships/outfits for sale in a system.

![Screenshot from 2023-07-19 10-39-26](https://github.com/endless-sky/endless-sky/assets/11161996/ffad4a40-8dc0-4ba0-8474-6aac9f987c26)

In most cases, it would be enough to squash the existing range of values by one and add an "unvisited" entry.  However, if there are multiple shipyards/outfitters in a system, and they have different items for sale, and the player only visits one, what should be displayed?  Perhaps the range entries could have a second color added (like the three colors for friendly/hostile), where the first color represents fully known info (the blue range), and the second partial info (the orange/red range).

![Screenshot from 2023-07-19 10-39-10](https://github.com/endless-sky/endless-sky/assets/11161996/52921063-7f62-4644-9aca-7f50da533880)

### Alternative Approaches

Not showing license-restricted items would help, as (so far as I've seen) the alien items tend to always have license restriction.  However, that's not guaranteed, and would also prevent showing things that should probably be displayed, like Navy ships.

### Additional Context

I'm posting this as a feature request rather than a PR because I'm not certain of the best way to solve the map issue.

**Comment by TomGoodIdea** [comment]
related: #8531

**Comment by flowers9** [comment]
Yep, I missed that one.  But I wouldn't want what that suggested either (having to step into the shipyard/outfitter to get the information), and my argument is more removing useless information that distracts the player, rather than attempting for more realism.  Encouraging the player to visit planets (where they can check out the spaceport for missions!) doesn't seem like a bad thing, play-wise.

**Comment by pepoluan** [comment]
I agree!

It's surprising to see a LOT of info 'leaked' just by visiting a system without even landing. And/or checking out the shipyards/outfitters.

I was originally going to suggest the same, i.e., don't show ships/outfits if one hasn't _at least_ landed on the planet on which the ship/outfit is sold.

**Comment by Bulinus** [comment]
I also agree.  I found it very strange that I suddenly had all this new information on alien tech I've never seen before in my map simple because I was exploring various systems with my jump drive.  I was unable to land in most of the systems but yet somehow I know what they sell and all the stats of their weapons.  In my opinion you should not be able to see this information without having landed on the planet or bought a map that includes the planet.

**Comment by User-green** [comment]
This would be a hassle.
Proposal: Reveal outfitters content info only for friendly, non-restricted ports.

**Comment by brendanjones** [comment]
>  I found it very strange that I suddenly had all this new information on alien tech I've never seen before in my map simple because I was exploring various systems with my jump drive.

Ditto to that.

> Proposal: Reveal outfitters content info only for friendly, non-restricted ports.

Logical in terms of realism (that info could easily be transmitted to you off-planet. You'd surely be getting info about a planet before you land on it, given your engineers would want to prepare for repairs, etc), but maybe not very intuitive to players? Not sure how you'd explain this mechanic or make it obvious that's what's happening.

**Comment by User-green** [comment]
> > Proposal: Reveal outfitters content info only for friendly, non-restricted ports.
> 
> Logical in terms of realism (that info could easily be transmitted to you off-planet. You'd surely be getting info about a planet before you land on it, given your engineers would want to prepare for repairs, etc), but maybe not very intuitive to players? Not sure how you'd explain this mechanic or make it obvious that's what's happening

- Add new color for outfitter map. ("Unknown")
- Add new dialog (say, on entering Hai space), that states smething along the lines of "I have trouble accesisng  their systems, so I have to go to spaceport in order to find a specialist, who can link them. 


Or, may be, it can be something like starmap, but only one per fraction?
